src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ITypeSymbolExtensions.TypeSyntaxGeneratorVisitor.cs

```diff
@@ -109,6 +110,21 @@ public override TypeSyntax VisitDynamicType(IDynamicTypeSymbol symbol)
                     SyntaxFactory.IdentifierName("dynamic"), symbol);
             }
 
+#if !CODE_STYLE
+
+            public override TypeSyntax VisitFunctionPointerType(IFunctionPointerTypeSymbol symbol)
+            {
+                // TODO(https://github.com/dotnet/roslyn/issues/39865): generate the calling convention once exposed through the API
+                var parameters = symbol.Signature.Parameters.Select(p => p.Type)
+                    .Concat(SpecializedCollections.SingletonEnumerable(symbol.Signature.ReturnType))
+                    .SelectAsArray(t => SyntaxFactory.Parameter(SyntaxFactory.MissingToken(SyntaxKind.IdentifierToken)).WithType(t.GenerateTypeSyntax()));
+
+                return AddInformationTo(
+                    SyntaxFactory.FunctionPointerType(SyntaxFactory.SeparatedList(parameters)), symbol);
```

Comment:
can you have a nullable function pointer type? i.e. `delegate*<int, int>?`?

Reply:
According to @333fred, no.

Reply:
Function pointer types cannot be type arguments, so you can't have a `Nullable<delegate*<void>>`